Source/ACE.Server/WorldObjects/Player_Inventory.cs

```diff
@@ -910,7 +910,9 @@ private bool DoHandleActionPutItemInContainer(WorldObject item, Container itemRo
                 }
             }
 
-            if (!container.TryAddToInventory(item, placement, true))
+            var burdenCheck = itemRootOwner != this && containerRootOwner == this;
```

Comment:
I think this should be:

var burdenCheck = itemRootOwner != this || containerRootOwner != this;

That way the code still does burden checks on the target container for items that exit the player.